### PR TITLE
[tensor_trainer] Add function to create a framework in the chain function

### PR DESCRIPTION
**Pull request by songgot**
A pushed gstbuffer by some src-element(not datareposrc) is prerolled if set sync=true.
so, it needs to create a framework in the chain function
- Add function to create a framework in the chain function
- Modify code related g_cond_wait

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [*]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: gst/nnstreamer/elements/gsttensor_trainer.c [hunk 1]
```diff
@@ -544,6 +548,31 @@ gst_tensor_trainer_epochs_is_complete (GstTensorTrainer * trainer)
...
+  g_mutex_lock (&trainer->framework_created_lock);
+  if (trainer->fw_created == TRUE) {
+    /* It's already created */
+    g_mutex_unlock (&trainer->framework_created_lock);
+    return;
+  }
+
+  GST_INFO_OBJECT (trainer, "wait for framework_created_cond signal");
+  g_cond_wait (&trainer->framework_created_cond,
+      &trainer->framework_created_lock);
```
Comment (gichan-jang) [inline gst/nnstreamer/elements/gsttensor_trainer.c:570]: How about adding while loop to check the if fw is created.
```suggestion
while (!trainer->fw_created)
  g_cond_wait (&trainer->framework_created_cond,
      &trainer->framework_created_lock);
```

Comment (songgot) [reply gst/nnstreamer/elements/gsttensor_trainer.c:570]: Thank you.!!

Comment (songgot) [reply gst/nnstreamer/elements/gsttensor_trainer.c:570]: Thank you.

**Review by gichan-jang — APPROVED** [review]
LGTM

**Comment by songgot** [comment]
Thank you for your review :+1:

**Review by again4you — APPROVED** [review]
LGTM!

**Review by jaeyun-jung — APPROVED** [review]

**Pull request merged**